src/map/packets/basic.h

```diff
@@ -134,7 +135,7 @@ class CBasicPacket
 
     std::size_t getSize()
     {
-        return (size_t)2 * (ref<uint8>(1) & ~1);
+        return std::min<std::size_t>((size_t)2 * (ref<uint8>(1) & ~1), PACKET_SIZE);
```

Comment:
While you're here can we kill this c-style cast? You can probably change the 2's and 1's to 2U and 1U to appease the compiler. Or make a helper so you can call 2_size, 1_size, etc.

Reply:
Should be done, waiting on CI to see if it hates what I did.